lib/std/event/loop.zig

```diff
@@ -796,10 +796,13 @@ pub const Loop = struct {
                 .waiters = DelayQueue.Waiters{
                     .entries = std.atomic.Queue(anyframe).init(),
                 },
-                .thread = try std.Thread.spawn(self, DelayQueue.run),
+                .thread = undefined,
                 .event = std.AutoResetEvent{},
-                .is_running = true,
+                .is_running = undefined,
             };
+
+            @atomicStore(bool, &self.is_running, true, .SeqCst);
```

Comment:
Does this need to be atomic? The fix is the fact that you start the thread after setting the field right?

Reply:
Isn't the compiler allowed to reorder operations unless explicitly stated otherwise through an atomic operation? I thought you always needed to "match" atomic reads and writes when there is concurrency.

Happy to be corrected though.

Reply:
Hmm, I think we will define `std.Thread.spawn` to have a memory fence, so you would not need to worry about this particular boundary. This is already implicitly the case based on the fact that it invokes a syscall.

You do need to match atomic reads and writes, but the `is_running = true` "happens before" the thread spawn, and the atomic read "happens after" the thread spawning. So with a memory fence at thread spawn, this is fine.

Reply:
What are you trying to order in relation to?

Reply:
@andrewrk done, thanks!

@daurnimator the `run` function checks that flag right after being spawned in a new thread.